### PR TITLE
Task - edit about me text

### PR DESCRIPTION
**Pull request by katestarks**

### CODE REVIEW

File: test/cms_index.php [hunk 1]
```diff
@@ -28,27 +28,154 @@ public function testDisplayAboutMeTextMalform()
...
-        $case = displayAboutMeText($aboutMeText);
+        $aboutMeQuote = [['thing'=>'words']];
+        $case = displayAboutMeQuote($aboutMeQuote);
         $this->assertEquals($expected, $case);
     }
 
     public function testDisplayAboutMeQuoteMalform()
     {
-        $aboutMeText = 'string';
+        $aboutMeQuote = 'string';
```
Comment (HazDuck) [inline test/cms_index.php]: change to input etc

File: test/cms_index.php [hunk 2]
```diff
@@ -28,27 +28,154 @@ public function testDisplayAboutMeTextMalform()
...
+
+    public function testCheckAddMeTextMalform()
+    {
+        $input = ['thing'];
+        $this->expectException(TypeError::class);
+        checkAddMeText($input);
+    }
+
+    public function testAddAboutMeSuccessFalse() {
+        $expected = "<p>Please add content of up to 400 characters</p>";
```
Comment (HazDuck) [inline test/cms_index.php:88]: functions () {
}

tests () 
{

}

this is the format that james does so i copied it :)

File: cms_index.php [hunk 3]
```diff
@@ -28,19 +49,32 @@
         <textarea class="typeText" name="addAboutMeText" form="addAboutMeText"></textarea>
         <input type="submit" value="Add text">
     </form>
+    <?php
+    echo $addAboutMeSuccess;
```
Comment (HazDuck) [inline cms_index.php]: put an isset() like charlie said for aldos code review

File: cms_index.php [hunk 4]
```diff
@@ -28,19 +49,32 @@
...
-        <input type="submit" value="Edit text">
+                <?php
+                if (isset($editId)) {
+                    echo '<input type="hidden" value=' . $editId . ' name="editId">';
+                    }
+                echo $displaySubmitEditButton;
+                ?>
     </form>
+    <?php
+    echo $editAboutMeSuccess;
```
Comment (HazDuck) [inline cms_index.php]: put an isset() like charlie said for aldos code review

**Review by HazDuck — COMMENTED** [review]
Also the issue that we spoke about before where i can edit an input and submit it as a blank

File: cms_index.php [hunk 5]
```diff
@@ -28,19 +49,32 @@
...
+            <?php
+            echo $displayEditAboutMeDropdown;
+            ?>
         </select>
         <input type="submit" value="select text">
     </form>
-    <form method="post">
-            <textarea class="typeText" name="editAboutMeText" form="editAboutMeText">
-
+    <form method="post" action="" id="editAboutMeTextAndQuote">
```
Comment (HazDuck) [inline cms_index.php]: apparently post should be POST so change all

File: test/cms_index.php (hunk 2 above)
Comment (katestarks) [reply test/cms_index.php:88]: Done

File: test/cms_index.php (hunk 1 above)
Comment (katestarks) [reply test/cms_index.php]: Done

File: cms_index.php (hunk 3 above)
Comment (katestarks) [reply cms_index.php]: Done

File: cms_index.php (hunk 4 above)
Comment (katestarks) [reply cms_index.php]: Removed since I had to chose between this and the page reload

File: cms_index.php (hunk 5 above)
Comment (katestarks) [reply cms_index.php]: Done

**Review by HazDuck — APPROVED** [review]

**Pull request merged**